opendcs-rest-api/src/test/java/org/opendcs/odcsapi/fixtures/TomcatServer.java

```diff
@@ -0,0 +1,154 @@
+	private final Tomcat tomcatInstance;
+	private final StandardContext context;
+	private Manager sessionManager;
+
+	/**
+	 * Setups the baseline for tomcat to run.
+	 *
+	 * @param baseDir     set to the CATALINA_BASE directory the build has setup
+	 * @param cdaWar      points to the actual WAR file to load
+	 * @param port        Network port to listen on
+	 * @param contextName url prefix to use, can be "/","/cwms-data","/spk-data"
+	 *                    etc
+	 * @throws Exception any error that gets thrown
+	 */
+	public TomcatServer(String baseDir, String cdaWar, int port, String contextName) throws Exception
+	{
+
+		tomcatInstance = new Tomcat();
+		tomcatInstance.setBaseDir(baseDir);
+		Host host = tomcatInstance.getHost();
+
+
+		host.setAppBase("webapps");
+		new File(tomcatInstance.getServer().getCatalinaBase(), "temp").mkdirs();
+		new File(tomcatInstance.getServer().getCatalinaBase(), "webapps").mkdirs();
+		tomcatInstance.setPort(port);
+		Connector connector = tomcatInstance.getConnector();
+		connector.setSecure(true);
+		connector.setScheme("https");
+		tomcatInstance.setSilent(false);
+		tomcatInstance.enableNaming();
+		Engine engine = tomcatInstance.getEngine();
+		logger.atTrace().log("Got engine " + engine.getDefaultHost());
+
+		host.addLifecycleListener(new HostConfig());
+		Pipeline pipeline = host.getPipeline();
+		pipeline.addValve(ssoValve);
+		context = (StandardContext) tomcatInstance.addWebapp(contextName, Paths.get("src/main/webapp").toAbsolutePath().toString());
+	}
+
+	public int getPort()
+	{
+		return tomcatInstance.getConnector().getLocalPort();
+	}
+
+	public Manager getTestSessionManager()
+	{
+		return this.sessionManager;
+	}
+
+	public SingleSignOnWrapper getSsoValve()
+	{
+		return this.ssoValve;
+	}
+
+	/**
+	 * Starts the instance of tomcat and returns when it's ready.
+	 *
+	 * @throws LifecycleException any error in the startup sequence
+	 */
+	public void start() throws LifecycleException
+	{
+		tomcatInstance.start();
+		this.sessionManager = context.getManager();
+		logger.atInfo().log("Tomcat listening at http://localhost:" + tomcatInstance.getConnector().getLocalPort());
+	}
+
+	/**
+	 * Used for the ./gradlew run command.
+	 * Unit tests only need to call start and move on.
+	 */
+	public void await()
+	{
+		tomcatInstance.getServer().await();
+	}
+
+	/**
+	 * Stops the instance of tomcat, including destroying the JNDI context.
+	 *
+	 * @throws LifecycleException any error in the stop sequence
+	 */
+	public void stop() throws LifecycleException
+	{
+		tomcatInstance.stop();
+	}
+
+	/**
+	 * arg[0] - the CATALINA_BASE directory you've setup
+	 * arg[1] - full path to the war file generated by this build script
+	 * arg[2] - name to use for this instance. See constructor for guidance
+	 *
+	 * @param args standard argument list
+	 */
+	public static void main(String[] args)
+	{
+		String baseDir = args[0];
+		String cdaWar = args[1];
```

Comment:
considering this was clearly taken directly from CDA, I wonder if we should just put it in its own project for convience?

Reply:
I couldn't get it working with the deployed war file like in CDA, having jaxrs jars on the tomcat class path as well as in the war libs sure makes it mad. After several failed attempts at cleaning up the path, I gave up and just left it out. Dunno if there's really enough common code for a shared project, but maybe?

Reply:
Fair, I can see there being some contention with jaxrs and tomcat. Not that perhaps there should be but why fight it unless we have to.